### PR TITLE
Feature - Manage payloads

### PR DESCRIPTION
**Pull request by jbaptperez**
This is the GUI part of the payload management, which depends on the [Caldera PR #2989](https://github.com/mitre/caldera/pull/2989).

* **Adds a payload view**:
  * Accessible from the navigation menu,
  * Handles payload file upload to `data/payloads` (and deletion),
  * Plugins payloads are excluded and left intact,
* **Optional**: Removes some white spaces in `Navigation.vue`,
* **Optional**: Excludes JavaScript generated files form the Docker context:
  * Lighten the Docker context when building a Caldera image, when some generated JavaScript files are present (local development),
  * Exclude such files as the Dockerfile handles itself the Vue.js file compilation.

### CODE REVIEW

**Comment by elegantmoose** [comment]
@jbaptperez reviewing and testing today. Apologies for delay.

File: src/components/core/Navigation.vue [hunk 1]
```diff
@@ -119,6 +121,7 @@ function promptToEnablePlugin(pluginName) {
             .dropdown-menu(role="menu")
                 .dropdown-content.ml-2
                     router-link.dropdown-item(to="/agents") agents
+                    router-link.dropdown-item(to="/payloads") payloads
```
Comment (elegantmoose) [inline src/components/core/Navigation.vue:124]: You think Payloads are a better fit in the "Campaigns" menu versus the "Configuration" menu?

Curious what @clenk  thinks.

**Comment by elegantmoose** [comment]
@jbaptperez  How hard would it be also show all the payloads loaded in the Caldera, and then have just the local ones the user uploaded have the the ability to delete?

*might be useful to show all the payloads (and their source plugins) in the table as well.

**Comment by elegantmoose** [comment]
Or we should make it clear on the payloads view, that it is ignoring all loaded payloads from plugins.

**Comment by elegantmoose** [comment]
Tested on Mac. All functionality works as described.

**Comment by jbaptperez** [comment]
> @jbaptperez How hard would it be also show all the payloads loaded in the Caldera, and then have just the local ones the user uploaded have the the ability to delete?
> 
> *might be useful to show all the payloads (and their source plugins) in the table as well. Or we should make it clear on the payloads view, that it is ignoring all loaded payloads from plugins.

@elegantmoose, I made the changes you asked for.

- I updated the payload view description to explicitly say that only local payloads can be uploaded or deleted (vs. plugin payloads),
- I split the table into 2 tables: Local editable payloads and plugin payloads (read-only). Only payloads of enabled plugins appear in the second table.

File: src/components/core/Navigation.vue (hunk 1 above)
Comment (elegantmoose) [reply src/components/core/Navigation.vue:124]: Yea I think I prefer it under "Configuration", if thats okay

File: src/views/PayloadsView.vue [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+const structuredPayloads = computed(() => {
+  const regex = /^(?:plugins\/([^/]+)|data)?\/payloads\/(.+)$/;
+  return payloads.value.map(payload => {
+    const match = payload.match(regex);
+    if (!match) {
+      console.error(`Payload path "${payload}" does not match the expected format.`);
+      return null;
+    }
+    let belongsToAPlugin = match[1] !== undefined;
```
Comment (elegantmoose) [inline src/views/PayloadsView.vue:24]: Just curious, why is this line needed? Isnt it just defined below in the returning dict?

**Comment by elegantmoose** [comment]
This LGTM, just that one new comment I left above, and then the move the payloads under the Configuration menu.

File: src/views/PayloadsView.vue (hunk 2 above)
Comment (jbaptperez) [reply src/views/PayloadsView.vue:24]: Good question.
As I said previously, I decided not to break the initial payload GET endpoint because it is being used by the ability view (when adding/modifying an ability).
This endpoint returns a list of strings, not a dictionary.
Note the swagger is wrong (**I can update it easily to the "true" returned data structure if you ask for it**).

Example of returned payloads:
```json
["payload1.exe", "payload2.sh"]
```

In the API, i just added _optional_ parameters that change the returned string (adding path, excluding plugins, sorting), but those parameters still do not return a proper list of dictionaries (which would be actually the ideal way to structure data).

So, at the frontend side, I managed to deduct (rebuild) such a structure, hence the code.

This is not ideal, but the GET payload endpoint breaking change could be handled in a different topic/issue/PR.

Just tell me what you prefer:

* validate this PR (no breaking change of the API)
* or request a change: Breaking GET payload API returning a list of structured data, implies less frontend boilerplate but the ability add/update view is to be updated.

**Comment by jbaptperez** [comment]
> This LGTM, just that one new comment I left above, and then the move the payloads under the Configuration menu.

Done and rebased.

I also fixed the upload modal width (insufficient when using files with long names) and, as I just discovered there were another existing upload modal in the adversaries view, I adapted the payload one for consistency.

File: src/views/PayloadsView.vue (hunk 2 above)
Comment (elegantmoose) [reply src/views/PayloadsView.vue:24]: Im okay with the current implementation if you are, and approve of merge. 

Im also okay with changing this in future set of PRs. *

**Comment by elegantmoose** [comment]
@jbaptperez  Thank you. 👏  Great PR.
